Purchases/Misc/Obsoletions.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Obsoletions.swift
+//
+//  Created by Nacho Soto on 11/15/21.
+
+import StoreKit
+
+// All these methods are `obsoleted`, which means they can't be called by users of the SDK,
+// and therefore the `fatalError`s are unreachable.
+
+public extension Purchases {
+
+    /**
+     * Deprecated
+     */
```

Comment:
it looks like these comment blocks are redundant now, could you remove them?

Reply:
They're public methods. We could disable the lint rule for this file I guess though. What do you think?

Reply:
Ohhh I see, that makes sense. 

Perhaps we could keep the old docs with a note explaining how they've been renamed? 
I'm thinking about the case where someone updates to RCv4, then Xcode tells them to update the method names and they might not be sure what the methods did in the first place. If they have the old docs in place with a note, it'll be clear, but otherwise they'll just run into a deprecated method that doesn't provide much context. 

What do you think? 

If you did decide to add the old docs, it can be done in a separate PR so as not to hold this one. 

So feel free to either keep this or disable the rule for this file for now, merge when ready, and we can continue the discussion about the docstrings in a follow-up PR. [Here are the old docs for reference](https://github.com/RevenueCat/purchases-ios/blob/3.13.0/Purchases/Public/RCPurchases.h).

Reply:
Thanks yeah that's definitely the right thing to do! I just added those.